### DISCUSSION TITLE
Grafana Dashboard

### DISCUSSION THREAD

**Discussion by lenaxia** [babybuddy/babybuddy#607] [Show and tell]
Hi all, the core dashboard wasn't quite cutting it for my needs, so I created a grafana dashboard that is better suited to operational use. 

Here's a screen shot:
![image](https://user-images.githubusercontent.com/6321838/222282622-fcd98b77-d6a2-4b36-ad8a-df7274af9f77.png)

I have this set up on an Amazon Fire Tablet 8 HD using Fully Kiosk with motion sensing so it lights up whenever my wife sits down to feed. This requires to be using the postgres backend for babybuddy.

It comes in handy because for instance, the time since last feed actually tells you that, AND time since you started feeding if you're mid feed, so you can track time. 

Of note:
* in the timeline, the different colors have meaning:
  * Feeding: orange is bottle feeding, pale yellow is breast feeding
  * diaper: red is solid, pink is wet, purple is both
  * sleeping: light blue is naps, dark blue is sleeping
* All counters count since last started (so feeding is since you last started feeding not since you last stopped feeding), except sleeping which tells you time since last sleep ended.
* the average by hour of day charts is what we used to help better understand our feeding patterns throughout the day.

You can get the dashboard here: https://github.com/lenaxia/home-ops-prod/blob/main/cluster/apps/home/babybuddy/dashboard/babybuddy.json

**Comment by cdubz** [comment]
Very cool! Thanks for sharing. I'll add the dashboard config link to [BBotW](https://github.com/babybuddy/babybuddy#baby-buddy-on-the-web).

**Comment by feranwq** [comment]
how to count the number of diapers used every week and every month? then thank you for your work @lenaxia

**Comment by smarinel** [comment]
Very interested in something like this, I am using Baby Scout to input data, but it doesn't update my iPad display automatically. It would be ideal to have a more real-time display. Not sure about how to use Grafana though?

**Comment by Mewks** [comment]
My instance of BabyBuddy is currently using the SQLLite database, is there any way to easily convert/migrate this to Postgres so I can connect Grafana to it?

**Reply by adampetrovic** [reply]
I managed to convert mine to postgres. It's a bit of a pain though. You'll need to drop certain database constraints (temporarily, then manually recreated them) that get created when the new database structure is bootstrapped in order for Django to populate the tables.

If you're up for that I can run you through it.

**Reply by th0mcat** [reply]
@adampetrovic if I ever get time to do a migration from SQLite to Postgres, I would love your assistance if you are still offering

**Comment by namelivia** [comment]
I'm setting up this one too, thank you for your work @lenaxia!

**Reply by lenaxia** [reply]
Here's an updated copy of the dashboard for 2.0

https://github.com/lenaxia/home-ops-prod/blob/main/cluster/apps/home/babybuddy/dashboard/babybuddy.json

**Comment by andremsousa** [comment]
Awesome job!
Do you have any guide on how to config BabyBuddy with Postgres in order use this dashboard?
Thanks!

**Reply by lenaxia** [reply]
@cdubz has great documentation already available here: https://docs.baby-buddy.net/configuration/database/. Use this as your DB_BACKEND value `django.db.backends.postgresql`

Beyond that, just configure grafana to point to your postgresdb, enter the username and password that you defined, and you should be able to import the dashboard I linked and update it to point to your configured postgres source. If you run into any issues feel free to ask specific questions.

**Comment by psilantropy** [comment]
Hi. I'd love to use this dashboard and make my own improvements to share, but need to migrate over to the new DB.

I started up a new app+db stack and used the database administration export and import functions to create CSVs. Unfortunately, towards the end I started hitting all these duplicate key problems on users then some other items.

I suspect maybe this was caused by me importing some activities before I remembered to create each user.

Is that export/import method an easy clickops way forward? It wasn't too time consuming to complete in this way.
I have completed the export/import task again, and it seems ok, but nervous about cutting over to the new instance now.

**Reply by psilantropy** [reply]
I think creating child/tags and users first helped and it looks ok. I'll probably export/import migrate soon.

**Reply by pierre-vo** [reply]
I did the migration. I might not have found the most efficient way to do it:
- create the users
- set the locale and timezone to match between the users used for export and import
- start with tags and children (take note of `child_id`)
- proceed with the rest of the export/import. I used _csv_ since I needed to manually add the `child_id` field.

Not a "one click procedure", but all in all it went fairly fast. Now happily running postgres and grafana :)

**Comment by psilantropy** [comment]
Made this based off my wife's usage.
It'll never be accurate (the surplus) due to wastage etc. But it's interesting anyway.

![image](https://github.com/babybuddy/babybuddy/assets/79955883/7ac835b1-ee75-4335-90f8-89f7a87bfc52)

I will definitely change this, as I think it can be presented far better. But it's a good first try at grafana. Appreciate the head start with the dashboard! <3
